File: src/model/pieces/queen.py
from src.model.pieces.piece import Piece


class Queen(Piece):
    def __init__(self, color, em):
        key_letter = "q"
        super().__init__(color, key_letter, em)
        self.moves = [
            ([0, 1], True),
            ([1, 1], True),
            ([1, 0], True),
            ([1, -1], True),
            ([0, -1], True),
            ([-1, -1], True),
            ([-1, 0], True),
            ([-1, 1], True)
        ]

File: src/pieces/queen.js
import Piece from '../piece.js'
import { bishopMovements } from './bishop.js'
import { rookMovements } from './rook.js'

export default class Queen extends Piece {
  constructor(color) {
    super(color, [...bishopMovements, ...rookMovements])
  }

  toString() {
    return this.color === 'black' ? '♕' : '♛'
  }
}
